api/v1/nificluster_types.go

```diff
@@ -445,6 +445,9 @@ type InternalListenerConfig struct {
 	Name string `json:"name"`
 	// The container port.
 	ContainerPort int32 `json:"containerPort"`
+	// +kubebuilder:validation:Enum={"TCP", "UDP", "SCTP"}
+	// The network protocol for this listener. Options defined here: https://pkg.go.dev/k8s.io/api/core/v1#Protocol
+	Protocol corev1.Protocol `json:"protocol,omitempty"`
```

Comment:
Just to be sure, can you add the default kubebuilder annotation (`+kubebuilder:default:=`). Although it seems that using an `Enum` takes the first element as default.

Reply:
Yes i can do, no problem!

Reply:
You are correct - kubebuilder defaults enums to the first value, but explicitly defining it is the appropriate approach.